.github/workflows/build-and-publish.yml

```
@@ -2,10 +2,12 @@ name: Build and Publish
 
 on:
   push:
+    branches:
+      - main
```

```suggestion
    branches: [ 'main' ]
    tags: [ '*' ]
```
I think this needs to include tags, otherwise we won't get image tags like `1.0.0`.